Changelog.md

```diff
@@ -11,6 +11,7 @@ Breaking Changes:
  * Type Checker: Disallow arithmetic operations for Boolean variables.
  * Disallow trailing dots that are not followed by a number.
  * Remove assembly instructions ``sha3`` and ``suicide``
+ * Code generator: revert if calldata has wrong size.
```

Comment:
We only revert if it is too short. Perhaps also mention that the revert happens inside the ABI decoder and this also applies to `abi.decode()` (where this has been the behaviour already before).